_studio/hevce_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -673,9 +673,10 @@ void VAAPIEncoder::FillSps(
     sps.general_profile_idc = par.m_sps.general.profile_idc;
     sps.general_level_idc   = par.m_sps.general.level_idc;
     sps.general_tier_flag   = par.m_sps.general.tier_flag;
-    sps.intra_period        = par.mfx.GopPicSize;
-    sps.intra_idr_period    = par.mfx.GopPicSize*par.mfx.IdrInterval;
-    sps.ip_period           = mfxU8(par.mfx.GopRefDist);
+    mfxU8 nPicturesPerFrame = par.isField() ? 2 : 1;
+    sps.intra_period        = par.mfx.GopPicSize * nPicturesPerFrame;
```

Comment:
Isn't Gop**Pic**Size and similars already in pictures (fields when interlace)?

Reply:
No. GopPicSize, GopRefDist and IdrInterval are specified in field pairs

Reply:
OK, except IdrInterval, which is unitless.